dev/com.ibm.ws.security.jwtsso_fat/fat/src/com/ibm/ws/security/jwtsso/fat/CookieProcessingTests.java

```diff
@@ -0,0 +1,395 @@
+            assertTrue("cookie jwtToken02 is not cleared", combinedCookieValues.contains("jwtToken02=\"\";"));
+        }
+    }
+
+    /**
+     * Test that after accessing a protected resource and getting back a token, a call to HttpServletRequest.logout()
+     * clears all jwt cookies
+     */
+    @Mode(TestMode.LITE)
+    @Test
+    public void test_ServletLogout() throws Exception {
+        reconfigServer("server_testlargecookies.xml");
+        doHappyPath();
+
+        // add attribute to tell the app to logout
+        String logoutUrl = protectedUrl + "?logout=true";
+        // now access resource a second time, force cookies to be rejoined into a single token
+        response = invokeUrl(wc, logoutUrl);
+        String responseStr = response.getWebResponse().getContentAsString();
+        boolean check2 = responseStr.contains("Test Application class BaseServlet logged out");
+        assertTrue("Did not get a response indicating logout was invoked", check2);
+        confirmCookiesCleared(true);
+
+        // and make sure we cannot access protected resource
+        response = invokeUrl(wc, protectedUrl);
+        responseStr = response.getWebResponse().getContentAsString();
+        assertFalse("should not have been able to access protected url ", responseStr.contains("SimpleServlet"));
+    }
+
+    /**
+     * Test that after accessing a protected resource and getting back a token, a call to the ibm_security_logout
+     * url clears all jwt cookies
+     */
+    @Mode(TestMode.LITE)
+    @Test
+    public void test_ibm_security_logout() throws Exception {
+        reconfigServer("server_testlargecookies.xml");
+        doHappyPath();
+        // add attribute to tell the app to logout
+        String logoutUrl = protectedUrl.replace("SimpleServlet", "ibm_security_logout");
+        // now access resource a second time, force cookies to be rejoined into a single token
+        response = invokeUrl(wc, logoutUrl, HttpMethod.POST);
+        confirmCookiesCleared(true);
+
+        // and make sure we cannot access protected resource
+        response = invokeUrl(wc, protectedUrl);
+        String responseStr = response.getWebResponse().getContentAsString();
+        assertFalse("should not have been able to access protected url ", responseStr.contains("SimpleServlet"));
+    }
+
+    /**
+     * Test that after accessing a protected resource and then logging out,
+     * Replaying the jwttoken is detected and access is denied.
+     */
+    @Mode(TestMode.LITE)
+    @Test
+    public void test_CookieReplay() throws Exception {
+        reconfigServer("server.xml");
+        doHappyPath();
+        String responseStr = response.getWebResponse().getContentAsString();
+        String beginStr = "cookie: jwtToken value: ";
+        int begin = responseStr.indexOf(beginStr) + beginStr.length();
+        int end = responseStr.indexOf("\n", begin);
+        String token = responseStr.substring(begin, end);
```

Comment:
As I mentioned the other day, you should be able to do `wc.getCookieManager().getCookie("jwtCookie")` to get the actual cookie object. You shouldn't have to parse the servlet output to get the cookie value.